sqlglot/dialects/clickhouse.py

```diff
@@ -205,12 +207,26 @@ def _parse_join_parts(
             )
 
         def _parse_join(self, skip_join_token: bool = False) -> t.Optional[exp.Join]:
```

Comment:
why don't you add ARRAY to JOIN_KINDS instead, this logic doesn't seem necessary